lib/spi/clone/CachingDirectoryManager.ts

```diff
@@ -110,26 +110,26 @@ type LockResult = LockAcquired | NoLockForYou;
 
 function pleaseLock(lockPath: string): Promise<LockResult> {
     return new Promise<LockResult>((resolve, reject) => {
```

Comment:
It looks like the new version of lockfile.lock supports returning a promise, so wrapping in a promise seems unnecessary and this method could be greatly simplified.

Reply:
I'll take a crack at revising this, or removing, as mentioned https://github.com/atomist/automation-client/pull/639#pullrequestreview-336622281